src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig

```diff
@@ -62,7 +62,22 @@
                 </div>
 
                 {% if templates is defined %}
+                    <script type="text/javascript">
+                        (function(Sfjs) {
+                            if ('true' == Sfjs.getPreference('sidebar/collapse')) {
+                                Sfjs.addClass(document.body, 'sidebar-collapse');
+                            }
+                        })(Sfjs);
+                    </script>
                     <ul id="menu-profiler">
+                        <li>
+                            <a href="#" onclick="javascript: Sfjs.toggleClass(document.body, 'sidebar-collapse'); Sfjs.setPreference('sidebar/collapse', Sfjs.hasClass(document.body, 'sidebar-collapse')); return false;">
```

Comment:
``javascript:`` in the onclick handler is wrong. You are confusing onclick and special href